packages/beacon-node/src/api/impl/beacon/blocks/index.ts

```diff
@@ -256,7 +256,7 @@ export function getBeaconBlockApi({
     }
 
     // TODO: Validate block
-    const delaySec = seenTimestampSec - (chain.genesisTime + slot * config.SECONDS_PER_SLOT);
+    const delaySec = seenTimestampSec - (chain.genesisTime + (slot * config.SLOT_DURATION_MS) / 1000);
```

Comment:
How is the slot duration being changed in a future fork? Update of `config.SLOT_DURATION_MS`? If so, then this calculation will be wrong.

I tend to think we need something like a `config.getTimeAtSlot` to abstract the calculation given differing slot durations across forks. (use anywhere you see a config.genesisTime being used, which is a red flag for calcuating the time at a slot)

Reply:
> How is the slot duration being changed in a future fork? Update of config.SLOT_DURATION_MS? If so, then this calculation will be wrong.

Assuming we change slot times in H* there would be another constant called `SLOT_DURATION_MS_H*`. I think the right way to deal with this is add a getter like `config.getSlotDurationMs(fork)`.

> I tend to think we need something like a config.getTimeAtSlot to abstract the calculation given differing slot durations across forks. (use anywhere you see a config.genesisTime being used, which is a red flag for calcuating the time at a slot)

yeah can clean those up too, I'd rather see it as part of `clock`, not `config`